content/docs/rfcs/47/README.md

```diff
@@ -0,0 +1,94 @@
+
+This Specification is a [free and open standard](http://www.digistan.org/open-standard:definition) and is governed by the Digital Standards Organization's [Consensus-Oriented Specification System](http://www.digistan.org/spec:1/COSS).
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+
+## Goals
+
+This specification is intended to formally document the names and expected behaviour of the CLIENT & SERVER socket types. Conforming implementations of these sockets SHOULD respect this specification, thus ensuring that applications can depend on predictable behavior. This specification is not transport specific, but not all behaviour will be reproducible on all transports.
+
+## Overall Goals of this Pattern
+
+Client-server pattern is member of a new family of thread-safe sockets.
+The client-server pattern is the thread-safe alternative of the router-dealer pattern.
+The client-server pattern is intended for service-oriented architectures of various kinds and it is asynchronous (like the router-dealer pattern).
+
+In order for the API to be thread-safe sending and receiving messages MUST be atomic and a single API call to receive or send entire msg. Therefore, the client-server pattern (and the rest of the thread-safe familty) MUST NOT allow multipart messages.
+
+## The CLIENT Socket Type
+
+The CLIENT socket type talks to a set of anonymous peers, sending and receiving messages using round-robin algorithms. It is reliable, insofar as it does not drop messages. 
+
+General behavior:
+
+* MAY be connected to any number of SERVER peers, and MAY both send and receive messages.
+* SHALL not filter or modify outgoing or incoming messages in any way.
+* SHALL maintain a double queue for each connected peer, allowing outgoing and incoming messages to be queued independently.
+* SHALL create a double queue when initiating an outgoing connection to a peer, and SHALL maintain the double queue whether or not the connection is established.
+* SHALL create a double queue when a peer connects to it. If this peer disconnects, the DEALER socket SHALL destroy its double queue and SHALL discard any messages it contains.
+* SHOULD constrain incoming and outgoing queue sizes to a runtime-configurable limit.
+
+For processing outgoing messages:
+
+* SHALL consider a peer as available only when it has a outgoing queue that is not full.
```

Comment:
Wording:  SHALL consider a peer as available only when *its outgoing queue is not full*.